From d92e05ea5ad46120ca38631bf21177090865e751 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Fri, 24 Apr 2020 10:59:34 +0800
Subject: [PATCH] ASoC: rk3308_codec: disable zero-cross without working AGC

Enabling zero-cross without AGC, it's may cause loss some
data when start capturing.

Change-Id: I57fcf2cdfc0f67b54700639c4a73bf6c92ac254a
Reported-by: XiaoTan Luo <lxt@rock-chips.com>
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 sound/soc/codecs/rk3308_codec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/rk3308_codec.c b/sound/soc/codecs/rk3308_codec.c
index edf1139fe78a..51346beb65aa 100644
--- a/sound/soc/codecs/rk3308_codec.c
+++ b/sound/soc/codecs/rk3308_codec.c
@@ -5108,7 +5108,7 @@ static int rk3308_platform_probe(struct platform_device *pdev)
 
 	rk3308->adc_grp0_using_linein = ADC_GRP0_MICIN;
 	rk3308->dac_output = DAC_LINEOUT;
-	rk3308->adc_zerocross = 1;
+	rk3308->adc_zerocross = 0;
 	rk3308->pm_state = PM_NORMAL;
 
 	platform_set_drvdata(pdev, rk3308);
-- 
2.35.3

